lib/common/options.js

```diff
@@ -64,18 +64,16 @@ Options.from = function (_options) {
   // Populate modulesDir
   options.modulesDir = []
     .concat(options.modulesDir)
-    .concat(path.join(options.nuxtDir, 'node_modules'))
-    .filter(hasValue)
+    .concat(path.join(options.nuxtDir, 'node_modules')).filter(hasValue)
     .map(dir => path.resolve(options.rootDir, dir))
 
-  // Sanitize extensions
-  if (!options.extensions.includes('js')) {
-    options.extensions.unshift('js')
-  }
+  const extensionsToRemoveIfPresent = ['js', 'mjs']
 
-  if (!options.extensions.includes('mjs')) {
-    options.extensions.unshift('mjs')
-  }
+  extensionsToRemoveIfPresent.forEach((ext) => {
+    if (!options.extensions.includes(ext)) {
+      options.extensions.unshift(ext)
```

Comment:
unlike `push`, every `unshift` causes a reindexing of all existing keys.
[concat](https://github.com/nuxt/nuxt.js/pull/3612#discussion_r215022110) would make this happen at most once.

Reply:
Why does it matter at all?

Reply:
@aldarund honor, mostly, in this case.

Reply:
@qm3ster It just feels like optimization for sake of optimization to show that we can optimize. Instead of solving real issues time is wasting on this. I know its fun and so, but ...

Reply:
Thanks for the illumination!

Reply:
@aldarund In this specific case I just felt like my filter -> concat is to this moment unsurpassed in readability.
At no point did I imply that my "use a loop so that the input array is only iterated once" solution should be considered for its performance superiority, because no one in history will ever input a list of even 10 extensions.
Maybe we should just do
```js
options.extensions.push('js','mjs') //because the order probably doesn't matter. They'd stay on the end anyway if someone put in ['coffee','ts','js','mjs']
options.extensions = _.uniq(options.extensions)
```
and forget about writing custom logic for this altogether.